### PR TITLE
feat: Add parameter `inplanes` and `nb_stages`

### PR DESCRIPTION
**Pull request by TheoRudkiewicz**
Allow to changes the number of channels and number of stages in the ResNet architecture

Also add the mask parameter in the extended_forward.

### CODE REVIEW

File: src/gromo/containers/resnet.py [hunk 1]
```diff
@@ -231,7 +236,9 @@ def forward(self, x: torch.Tensor) -> torch.Tensor:
         return x
 
     def extended_forward(  # pyright: ignore[reportIncompatibleMethodOverride]
-        self, x: torch.Tensor
+        self,
+        x: torch.Tensor,
+        mask: None = None,  # noqa: ARG002
```
Comment (alexdavey) [inline src/gromo/containers/resnet.py]: Is this supposed to be part of this PR?

File: src/gromo/containers/resnet.py [hunk 2]
```diff
@@ -295,6 +304,12 @@ def init_full_resnet_structure(
         Number of basic blocks per stage. If an integer is provided, the same number
         of blocks will be used for all four stages. If a tuple is provided, it should
         contain four integers specifying the number of blocks for each stage.
```
Comment (alexdavey) [inline src/gromo/containers/resnet.py]:
```suggestion
        of blocks will be used for all stages. If a tuple is provided, it should
        contain `nb_stages` integers specifying the number of blocks for each stage.
```

File: src/gromo/containers/resnet.py [hunk 3]
```diff
@@ -317,20 +332,22 @@ def init_full_resnet_structure(
...
-        and len(number_of_blocks_per_stage) == 4
+        and len(number_of_blocks_per_stage) == nb_stages
     ):
         blocks_per_stage = number_of_blocks_per_stage
     elif isinstance(number_of_blocks_per_stage, int):
-        blocks_per_stage = (number_of_blocks_per_stage,) * 4
+        blocks_per_stage = (number_of_blocks_per_stage,) * nb_stages
     else:
         raise TypeError(
             "number_of_blocks_per_stage must be an int or a tuple of four ints."
```
Comment (alexdavey) [inline src/gromo/containers/resnet.py]:
```suggestion
            f"number_of_blocks_per_stage must be an int or a tuple of {nb_stages} ints."
```

File: src/gromo/containers/resnet.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (TheoRudkiewicz) [reply src/gromo/containers/resnet.py]: Yes. Anything else will not work.

File: src/gromo/containers/resnet.py [hunk 4]
```diff
@@ -91,14 +99,13 @@ def __init__(
...
         self.stages: nn.ModuleList = nn.ModuleList()
-        nb_stages = 4
         for i in range(nb_stages):
             # for the future we could remove the basic block of the first stage
             # as there is no dowsampling
             stage = nn.Sequential()
             input_channels = inplanes * (2 ** max(0, i - 1))
             output_channels = inplanes * (2**i)
-            hidden_channels = int(inplanes * (2**i) * self.reduction_factor)
+            hidden_channels = ceil(inplanes * (2**i) * self.reduction_factor)
```
[the remark this answers is not in the document]
Comment (TheoRudkiewicz) [reply src/gromo/containers/resnet.py]: It's ok.
